"No module named 'distutils'" with old setuptools on Python 3.12

On Python 3.12, when running tests with old versions of setuptools (using `--force-dep` in tox), specifically version 56, the pre-testing setup fails with this error:

```console
$ tox -r -e py312 --force-dep setuptools==56
[...]
py312: commands[0]> pytest tests/test_classifiers.py
ImportError while loading conftest '/home/diazona/programming/setuptools-pyproject-migration/tests/conftest.py'.
tests/conftest.py:4: in <module>
    import test_support
test_support/test_support/__init__.py:1: in <module>
    import distutils.core
E   ModuleNotFoundError: No module named 'distutils'
py312: exit 4 (0.40 seconds) /home/diazona/programming/setuptools-pyproject-migration> pytest tests/test_classifiers.py pid=31540
```

Discovered in [this example in CI](https://github.com/diazona/setuptools-pyproject-migration/actions/runs/7167025692/job/19512272944).

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -80,6 +80,14 @@ jobs:
         # Note the -- used to separate tox options from pytest options.
         run: >
           tox -e py
-          ${{ inputs.force-minimum-dependencies && '--force-dep setuptools==56 --force-dep setuptools_scm==3.4.1 --force-dep typing-extensions==4' || '' }}
+          ${{
+            inputs.force-minimum-dependencies
+            && (
+              inputs.python-version >= '3.12'
+              && '--force-dep setuptools==66.1 --force-dep setuptools_scm==3.4.1 --force-dep typing-extensions==4'
+              || '--force-dep setuptools==56 --force-dep setuptools_scm==3.4.1 --force-dep typing-extensions==4'
+            )
+            || ''
+          }}
           --
           ${{ !inputs.run-slow && '-m "not slow"' || '' }}

--- a/changelog.d/139.bugfix.rst
+++ b/changelog.d/139.bugfix.rst
@@ -1,0 +1,1 @@
+Require ``setuptools>=66.1`` when running on Python 3.12

--- a/setup.cfg
+++ b/setup.cfg
@@ -30,7 +30,8 @@ python_requires = >=3.6
 install_requires =
 	packaging
 	pep508-parser
-	setuptools
+	setuptools; python_version < "3.12"
+	setuptools >= 66.1; python_version >= "3.12"
 	tomlkit
 	typing-extensions >=4, <5
 

